base/src/org/compiere/process/InvoiceCreateFrom.java

```diff
@@ -112,22 +112,7 @@ protected String doIt() throws Exception {
 				MOrderLine orderLine = new MOrderLine (getCtx(), key, get_TrxName());
 				//	Set reference
 				referenceId.set(orderLine.getC_Order_ID());
-				//	Set InOut
-				String whereClause = "EXISTS (SELECT 1 "
-						+ "FROM M_InOut io "
-						+ "WHERE io.M_InOut_ID = M_InOutLine.M_InOut_ID "
-						+ "AND io.DocStatus IN ('CO','CL'))";
-				MInOutLine[] inOutLines = MInOutLine.getOfOrderLine(Env.getCtx(), key, whereClause, get_TrxName());
-				log.fine ("Receipt Lines with OrderLine = #" + inOutLines.length);
```

Comment:
This code allows allocate receipt from order when a order have receipt allocated.

How to think you resolve this?

Reply:
The invoice match with purchase order was designed to validate the invoiced quantity with respect to the ordered quantity, so it should not be associated with a material receipt, for that it is the correspondence of receipt of materials with purchase order.

So these lines of code cause bad behavior.

The logic assigns the first receipt of materials that it finds even when it already has a previous match, in the case of partial or multiple receipts it generates the match for the same receipt of material causing a big problem with the matches, having to fix this manually which implies time and confusion in users

Reply:
Please put here a step by step for replicate it. This funcionality is legacy and has very little changes. I want to be sure of it